src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs

```diff
@@ -99,21 +99,22 @@ protected override bool ReceiveCommand(object message)
                     }
                     break;
                 case SaveSnapshotFailure m:
-                    Log.Warning("PersistentShard snapshot failure: {0}", m.Cause.Message);
+                    Log.Warning("PersistentShard snapshot failure: [{0}]", m.Cause.Message);
                     break;
                 case DeleteMessagesSuccess m:
-                    Log.Debug("PersistentShard messages to {0} deleted successfully", m.ToSequenceNr);
-                    DeleteSnapshots(new SnapshotSelectionCriteria(m.ToSequenceNr - 1));
+                    var deleteTo = m.ToSequenceNr - 1;
+                    var deleteFrom = Math.Max(0, deleteTo - Settings.TunningParameters.KeepNrOfBatches * Settings.TunningParameters.SnapshotAfter);
+                    Log.Debug("PersistentShard messages to [{0}] deleted successfully. Deleting snapshots from [{1}] to [{2}]", m.ToSequenceNr, deleteFrom, deleteTo);
+                    DeleteSnapshots(new SnapshotSelectionCriteria(deleteTo, DateTime.MaxValue, deleteFrom));
```

Comment:
maxTimeStamp is not required in the JVM. Are we missing a SnapshotSelectionCriteria overload here?

Reply:
@ismaelhamed I think so - mind submitting a separate PR for that?